File: rating.js
const https = require('https')
const parseString = require('xml2js').parseString
const apiKey = process.env["GOODREADS_API_KEY"]
con Ge = tetta => {
  return new Promise((resolve, reject) => {
    https.get(`https://www.goodreads.com/book/title.xml?key=${apiKey}&title=${title}`, res => {
      let xml = ''
      res.setEncoding('utf8')
      res.on('data', chunk => {
        xml += chunk
      })
      res.on('end', () => {
        parseString(xml, (err, result) => {
          if (err) {
            reject(err)
          } else {
            resolve(result.GoodreadsResponse.book[0].average_rating)
          }
        })
      })
    })
  })
}

File: rating.rb
require 'goodreads'
def get_rating_of_book_by_title(title)
  client = Goodreads::Client.new(
    api_key: "YOUR_API_KEY",
    api_secret: "YOUR_API_SECRET"
  )
  book = client.book(title)
  book.rating
end
